cmd/otel-agent/subcommands/run/command.go

```diff
@@ -170,19 +184,39 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 		// ctx is required to be supplied from here, as Windows needs to inject its own context
 		// to allow the agent to work as a service.
 		fx.Provide(func() context.Context { return ctx }), // fx.Supply(ctx) fails with a missing type error.
+
+		// TODO: consider adding configsync.Component as an explicit dependency for traceconfig
+		//       to avoid this sort of dependency tree hack.
+		fx.Provide(func(deps traceconfig.Dependencies, _ optional.Option[configsync.Component]) (traceconfig.Component, error) {
+			// TODO: this would be much better if we could leverage traceconfig.Module
+			//       Must add a new parameter to traconfig.Module to handle this.
+			return traceconfig.NewConfig(deps)
+		}),
+		fx.Supply(traceconfig.Params{FailIfAPIKeyMissing: false}),
+
 		fx.Supply(&traceagentcomp.Params{
 			CPUProfile:  "",
 			MemProfile:  "",
 			PIDFilePath: "",
 		}),
-		tracecomp.Bundle(),
+		traceagentfx.Module(),
 	)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func newForwarderParams(config coreconfig.Component, l log.Component) defaultforwarder.Params {
-	return defaultforwarder.NewParams(config, l)
+// TODO: cleanup the forwarder instantiation with fx.
+// This is a bit of a hack because we need to enforce optional.Option[configsync.Component]
+// is passed to newForwarder to enforce the correct instantiation order. Currently, the
+// new forwarder.BundleWithProvider makes a few assumptions in its generic prototype, and
+// this is the current workaround to leverage it.
```

Comment:
@ogaca-dd @truthbk we might have some conflicts depending on which PR gets merge first 
https://github.com/DataDog/datadog-agent/pull/28862

Reply:
I hope I win this race. 😂